### PR TITLE
Fix conda virtual environment paths in notebooks

### PR DESCRIPTION
**Pull request by corivera**
When using Anaconda, there's a python executable in each virtual environment's folder, which can be used in our Configure Python dialog to run notebooks with. Currently conda packages can't be installed when using such a path because our notebooks fallback to using just pip if conda can't be found in the provided path. This PR fixes that issue by adding a check for environment folders if the initial check for the root folder fails. There's a specific "envs" folder in the anaconda folder, so we just need to check 2 folders above the virtual environment folder when looking for the conda executable. Example: Anaconda3\envs\myEnv\python.exe -> Anaconda3\conda.exe

Fixes https://github.com/microsoft/azuredatastudio/issues/20239

### CODE REVIEW

**Comment by corivera** [comment]
Note that conda environments still need to manually activated before opening ADS. This just allows the env paths to be usable in the first place.

**Comment by corivera** [comment]
Smoke test failure is another QueryEditor failure, so unrelated to this change.

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts [hunk 1]
```diff
@@ -714,9 +711,28 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
...
-			process.platform === constants.winPlatform ? 'Scripts\\conda.exe' : 'bin/conda');
+	public get condaExecutable(): string | undefined {
+		return this._condaExecutable;
+	}
+
+	public static async getCondaExePath(pythonInstallationPath: string): Promise<string> {
+		let exeName = process.platform === constants.winPlatform ? 'Scripts\\conda.exe' : 'bin/conda';
+		let condaPath = path.join(pythonInstallationPath, exeName);
+		let condaExists = await fs.pathExists(condaPath);
+		// If conda was not found, then check if we're using a virtual environment
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/jupyterServerInstallation.ts:722]: Would suggest adding a few logging statements here for the searching to help with debugging further issues with conda environments.

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts [hunk 2]
```diff
@@ -714,9 +711,28 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
...
+		if (!condaExists) {
+			let pathParts = pythonInstallationPath.split(path.sep);
+			if (pathParts.length > 1 && pathParts[pathParts.length - 2] === 'envs') {
+				// The root Anaconda folder is 2 folders above the virtual environment's folder
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/jupyterServerInstallation.ts:726]: Is this documented somewhere? If so would be good to add a link to that

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts (hunk 2 above)
Comment (corivera) [reply extensions/notebook/src/jupyter/jupyterServerInstallation.ts:726]: Added a link to the docs

**Review by MaddyDev — APPROVED** [review]

**Review by VasuBhog — APPROVED** [review]

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts (hunk 1 above)
Comment (corivera) [reply extensions/notebook/src/jupyter/jupyterServerInstallation.ts:722]: These path issues tend to be pretty obvious. I also don't want to pollute the log with "conda not found" messages for people using regular python.

**Comment by corivera** [comment]
The only CI test failure was a markdown test timeout, which is unrelated to this change. Tests passed locally.

**Pull request merged**